src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -475,6 +483,9 @@ private async Task EnqueueWorkItemAsync(Project project, InvocationReasons invoc
             {
                 foreach (var documentId in project.DocumentIds)
                     await EnqueueWorkItemAsync(project, documentId, document: null, invocationReasons).ConfigureAwait(false);
+
+                foreach (var document in await project.GetSourceGeneratedDocumentsAsync(CancellationToken.None).ConfigureAwait(false))
```

Comment:
❗ This is a non-cancellable call to run source generators on an IDE fast path

Reply:
➡️ The `_shutdownToken` is now passed through. Resolved by gating this on `WorkspaceConfigurationOptions.ShouldConnectSourceGeneratedFilesToWorkspace`